### PR TITLE
Move InstallButton into core

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/756

### CODE REVIEW

**Comment by kumar303** [comment]
@muffinresearch r? I wasn't sure if all these constants should live in core or not.

File: src/core/constants.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// Unrecoverable errors.
+export const FATAL_INSTALL_ERROR = 'FATAL_INSTALL_ERROR';
+export const FATAL_UNINSTALL_ERROR = 'FATAL_UNINSTALL_ERROR';
+export const FATAL_ERROR = 'FATAL_ERROR';
+
+// Add-on types.
+export const API_THEME_TYPE = 'persona';
+export const EXTENSION_TYPE = 'extension';
+export const THEME_TYPE = 'theme';
+// These types are not used.
```
Comment (muffinresearch) [inline src/core/constants.js]: We should probably remove these.

File: src/core/components/InstallButton/InstallButton.scss [hunk 2]
```diff
@@ -220,7 +220,7 @@ $installStripeColor2: #00c42e;
   &.enabled,
   &.installed {
     input + label::before {
-      background: $switchBackgroundOn url('../img/tick.svg') no-repeat 35% 50%;
+      background: $switchBackgroundOn url('./img/tick.svg') no-repeat 35% 50%;
```
Comment (muffinresearch) [inline src/core/components/InstallButton/InstallButton.scss]: I wonder if it's worth moving this image? This could possibly be re-used but maybe not.

Comment (kumar303) [inline src/core/components/InstallButton/InstallButton.scss]: This doesn't appear to be used by any other component. If we need to use it later on, we could put it somewhere in a `core/img` folder or something.

**Comment by muffinresearch** [comment]
r+ looks good.

It'll be worth testing disco pane on -dev just to make sure everything's still happy. See https://wiki.mozilla.org/Add-ons/Projects/DiscoveryImprovements#Testing for instructions.

**Comment by kumar303** [comment]
@muffinresearch thanks for that link. I set `extensions.webservice.discoverURL` to `http://127.0.0.1:3000/%LOCALE%/firefox/discovery/pane/%VERSION%/%OS%/%COMPATIBILITY_MODE%` on Nightly but I still get:

    _mozAddonManager is undefined

Are there additional steps for testing against my local server?

**Comment by muffinresearch** [comment]
You'll only be able to test against -dev / stage with those instructions. Anything else requires spoofing hosts. Which you can do with charles.

**Comment by kumar303** [comment]
we're going to land this on dev and see how it goes!

**Comment by kumar303** [comment]
I tested it on dev, seemed to work!

**Pull request merged**